test/integration/auth.spec.ts

```diff
@@ -833,53 +833,120 @@ describe('admin.auth', () => {
       });
   });
 
-  it('verifyIdToken() fails when called with an invalid token', () => {
-    return admin.auth().verifyIdToken('invalid-token')
-      .should.eventually.be.rejected.and.have.property('code', 'auth/argument-error');
-  });
+  describe('verifyIdToken()', () => {
+    const uid = generateRandomString(20).toLowerCase();
+    const email = uid + '@example.com';
+    const password = 'password';
+    const userData = {
+      uid,
+      email,
+      emailVerified: false,
+      password,
+    };
+
+    // Create the test user before running this suite of tests.
+    before(() => {
+      return admin.auth().createUser(userData);
+    });
+
+    // Sign out after each test.
+    afterEach(() => {
+      return clientAuth().signOut();
+    });
+
+    after(() => {
+      return safeDelete(uid);
+    });
 
-  if (authEmulatorHost) {
-    describe('Auth emulator support', () => {
-      const uid = 'authEmulatorUser';
-      before(() => {
-        return admin.auth().createUser({
-          uid,
-          email: 'lastRefreshTimeUser@example.com',
-          password: 'p4ssword',
+    it('verifyIdToken() fails when called with an invalid token', () => {
+      return admin.auth().verifyIdToken('invalid-token')
+        .should.eventually.be.rejected.and.have.property('code', 'auth/argument-error');
+    });
+
+    it('verifyIdToken() fails with checkRevoked set to true and corresponding user disabled', () => {
+      let currentIdToken: string;
+      return clientAuth().signInWithEmailAndPassword(email, password)
+        .then((result) => {
+          expect(result.user).to.exist;
+          expect(result.user!.email).to.equal(email);
+          return result.user!.getIdToken()
+        })
+        .then((idToken) => {
+          currentIdToken = idToken;
+          return admin.auth().verifyIdToken(currentIdToken, true)
+        })
+        .then((decodedIdToken) => {
+          expect(decodedIdToken.uid).to.equal(uid);
+          expect(decodedIdToken.email).to.equal(email);
+          return admin.auth().updateUser(uid, { disabled: true })
+        })
+        .then((userRecord) => {
+          // Ensure disabled field has been updated.
+          expect(userRecord.uid).to.equal(uid);
+          expect(userRecord.email).to.equal(email);
+          expect(userRecord.disabled).to.equal(true);
+          return clientAuth().signInWithEmailAndPassword(email, password)
```

Comment:
Shouldn't this fail now that the user account is disabled?

Reply:
Yes, I'll add a test case for updating a user by setting UserRecord.disabled = true, then signin get rejected.

For here, I think the logic is as expected that this shouldn't fail, because we've signed in once and get an 1-hour token, the 2nd signInWithEmail isn't the real 'sign in', but like a refresh, I need to do this to detect a second-call on verifyIdToken returns error.

Reply:
An example of 2nd calling signInWithEmailAndPassword doing refresh [here](https://github.com/firebase/firebase-admin-node/blob/ae944f4d083a7087224073711bf7fbc0bac84da1/test/integration/auth.spec.ts#L471)

Reply:
In that case can we just skip this step entirely, and just do the following?

```
return admin.auth().verifyIdToken(currentIdToken, true)
```

Then it's clear we are re-verifying the same ID token as before, but this time it's expected to fail.

Reply:
It makes sense to me that the 2nd verifyIdTokens should fail without a second sign-in, however it doesn't, which doesn't make sense(I understand in Python, that is how it works). I haven't found the reason why it doesn't work, all I find out is the similar example linked above(calling sign in twice), so I suspect there's a refresh needed. It could be hard for me to find another reason explaining that.

Reply:
I think `return clientAuth().signInWithEmailAndPassword(email, password)` will throw `auth/user-disabled` error. It will jump the next then statements until it reaches the error block. This will also pass since the client op will fail with `auth/user-disabled`. Try it out:
```
return clientAuth().signInWithEmailAndPassword(email, password)
  .then(() => {
    console.log('1');
  })
  .then(() => {
    console.log('2');
  })
  .then(() => {
    throw new Error('Unexpected success');
  }, (error) => {
    console.log(error.code);
  });
```

It will not print '1' or '2' and just print the error code

Reply:
I remove the 2nd signInWithEmailAndPassword, and using currentUser.reload() to make sure 2nd call verifyIdToken after modification will fail(otherwise it won't).